modules/openapi-generator/src/main/resources/python/api_client.mustache

```diff
@@ -75,7 +76,13 @@ class ApiClient(object):
         self.user_agent = '{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}OpenAPI-Generator/{{{packageVersion}}}/python{{/httpUserAgent}}'
         self.client_side_validation = configuration.client_side_validation
 
-    def __del__(self):
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def close(self):
         if self._pool:
             self._pool.close()
             self._pool.join()
```

Comment:
after the pool is closed and joined we should unregister the atexit -  https://docs.python.org/3/library/atexit.html#atexit.unregister; this is so that we don't keep the object in memory until the application dies

Reply:
:+1: Makes sense